feature-libs/cart/saved-cart/core/store/effects/saved-cart.effect.ts

```diff
@@ -0,0 +1,68 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import {
+  CartActions,
+  GlobalMessageService,
+  GlobalMessageType,
+  normalizeHttpError,
+  OCC_USER_ID_ANONYMOUS,
+} from '@spartacus/core';
+import { ClearCheckoutService } from 'projects/core/src/checkout/services/clear-checkout.service';
+import { Observable, of } from 'rxjs';
+import { catchError, filter, map, switchMap } from 'rxjs/operators';
+import { SavedCartConnector } from '../../connectors/saved-cart.connector';
+import { SavedCartActions } from '../actions/index';
+
+@Injectable()
+export class SavedCartEffects {
+  @Effect()
+  saveCart$: Observable<
+    | SavedCartActions.SaveCartSuccess
+    | SavedCartActions.SaveCartFail
+    | CartActions.ClearCartState
+  > = this.actions$.pipe(
+    ofType(SavedCartActions.SAVE_CART),
+    map((action: SavedCartActions.SaveCart) => action.payload),
+    filter((payload) => payload.userId !== OCC_USER_ID_ANONYMOUS),
+    switchMap(({ userId, cartId, cartDescription, cartName }) => {
+      return this.savedCartConnector
+        .create(userId, cartId, cartDescription, cartName)
+        .pipe(
+          switchMap((cart) => {
+            this.showSaveCartMessage(
+              'savedCartCartPage.messages.cartSaved',
```

Comment:
remove this from here as you can use the process state as mentioned in ? 

Basically, in your component, you hook into the success flag and you would fire this global message and reset the process state


example for your effects: https://github.com/SAP/spartacus/pull/11326/files#diff-d76cd55a98b1f476b0de586d875c33f1e3947e8eb57fc1787e9804b6bacb9e1fR126